File: Java/FileSystem/File.java

public class File extends AbstractFileSystemNode {

    public File(String inName, AbstractFileSystemNode inParent) {
        setName(inName);
        if (inParent instanceof Folder) {
            setParent(inParent);
            Folder folderParent = (Folder) inParent;
            folderParent.getChilds().add(this);
        } else {
            setParent(null);
        }
    }

    public String getExtension() {
        int indexOfExtension = getName().indexOf('.');
        return getName().substring(indexOfExtension + 1);
    }

}

File: Kotlin/FileSystemOnKotlin/src/main/kotlin/File.kt
class File() : AbstractFileSystemNode() {
    constructor(inName: String, inParent: AbstractFileSystemNode?) : this() {
        name = inName
        if (inParent is Folder) {
            parent = inParent
            inParent.childs.add(this)
        } else {
            parent = inParent
        }
    }

    fun getExtension(): String {
        return name.substring(name.lastIndexOf('.'))
    }
}
